### PR TITLE
feature: allow multiple aws providers

### PR DESCRIPTION
**Pull request by edulop91**
### Summary
We want to be able to configure multiple AWS providers, for multiple accounts and regions within a same component. This code introduces a new configuration option to the aws provider block that allows us to do so. I know this config structure is not ideal and kind of painful but seems like an ok compromise until the time we decide to introduce a new config version.

```yaml
accounts:
  bar:
    providers:
      aws:
        account_id: 456
        additional_regions:
          - us-east-1
          - us-east-2
        role: foo
        additional_providers:
          another_account_different_role:
            account_id: 789
            role: different_role
          another_account_same_role:
            account_id: different_account
          another_account_more_regions:
            account_id: another_different_account
            additional_regions:
              - eu-west-2
```
produces:
```hcl
provider aws {
  alias  = "another_account_different_role"
  region = "us-west-2"


  assume_role {
    role_arn = "arn:aws:iam::789:role/different_role"
  }

  allowed_account_ids = [789]
}


provider aws {
  alias  = "another_account_different_role-us-east-1"
  region = "us-east-1"


  assume_role {
    role_arn = "arn:aws:iam::789:role/different_role"
  }

  allowed_account_ids = [789]
}
.....
.....
```
### Test Plan
unittests

### CODE REVIEW

File: plan/plan.go [hunk 1]
```diff
@@ -516,40 +516,83 @@ func (p *Plan) buildEnvs(conf *v2.Config) (map[string]Env, error) {
...
-			Profile:   awsConfig.Profile,
-			Region:    *awsConfig.Region,
-			RoleArn:   roleArn,
+		// set so not nil
+		pseudoCommon.Providers.AWS.AdditionalProviders = map[string]*v2.AWSProvider{}
+
+		extraCommons := append(commons[:], pseudoCommon)
+		extraPlan, extraProviders, _ := resolveAWSProvider(extraCommons...)
+		if extraPlan != nil {
+			extraPlan.Alias = util.JoinStrPointers("-", &aliasPrefix, extraPlan.Alias)
```
Comment (cziie) [inline plan/plan.go:576]: 🚫 **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Using a reference for the variable on range scope `aliasPrefix` (scopelint)

File: plan/plan.go [hunk 2]
```diff
@@ -516,40 +516,83 @@ func (p *Plan) buildEnvs(conf *v2.Config) (map[string]Env, error) {
...
+			providers = append(providers, *extraPlan)
 		}
 
+		for _, eP := range extraProviders {
+			extraProvider := eP
+			extraProvider.Alias = util.JoinStrPointers("-", &aliasPrefix, extraProvider.Alias)
```
Comment (cziie) [inline plan/plan.go:582]: 🚫 **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Using a reference for the variable on range scope `aliasPrefix` (scopelint)

**Comment by kuannie1** [comment]
wait, do we need this change?

**Review by kuannie1 — APPROVED** [review]

**Comment by kuannie1** [comment]
seems like there's a failing test. Do you have the bandwidth to look at the test & determine how to fix/tweak it @edulop91 ?

**Comment by jgadling** [comment]
Just a note about the behavior here, not sure if it's expected:

```
    providers:
      aws:
        account_id: '123456'
        additional_providers:
          other-account:
            account_id: '789012'
        additional_regions:
          - ap-south-1
          - eu-west-3
```

This produces a set-product of accounts and additional_regions. So both 123456 and 789012 will have providers created for the default region, ap-south-1, and eu-west-3

Whereas this only creates a provider for 123456 in ap-south-1 and eu-west-3:

```
      aws:
        account_id: '123456'
        additional_providers:
          other-account:
            account_id: '789012'
            additional_regions: []
        additional_regions:
          - ap-south-1
          - eu-west-3

```

File: plan/plan.go [hunk 3]
```diff
@@ -175,6 +176,23 @@ type AWSProvider struct {
 	RoleArn   *string     `yaml:"role_arn"`
 }
 
+func (a *AWSProvider) String() string {
+	accountId := a.AccountID.String()
```
Comment (cziie) [inline plan/plan.go:180]: 🚫 **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>var `accountId` should be `accountID` (golint)

**Pull request merged**